enterprise/cli/provisionerdaemonstart_test.go

```diff
@@ -294,6 +294,51 @@ func TestProvisionerDaemon_ProvisionerKey(t *testing.T) {
 		require.Equal(t, proto.CurrentVersion.String(), daemons[0].APIVersion)
 	})
 
+	t.Run("OKWithTags", func(t *testing.T) {
```

Comment:
We could add a test to validate what happen in case of the `client.GetProvisionerKey` failing - I've just done it. 

Otherwise the logic should not be impacted and the endpoint itself seems pretty much well tested.

Reply:
I'm not really understanding the point of `NoProvisionerKeyFound`. The purpose of this PR is to add changes to tags which are logged, but AFAICS you're not looking at the outputted logs? Does the provisioner fail to start in this case?

Reply:
Yeah that's exactly it - I am open to the behavior we want to apply here but globally if we can not contact the backend / retrieve the key , I returned and error and did not continued to start the provisioner.

We can instead just output an error and continue as if nothing happened - but the log will be the same as of now (without the tags)

Reply:
Cool, no need to change the behaviour - I was just confirming what the intent was for this test.